Explorer/Assets/DCL/Infrastructure/Global/Dynamic/MainSceneLoader.cs

```diff
@@ -301,6 +301,39 @@ await bootstrap.InitializeFeatureFlagsAsync(bootstrapContainer.IdentityCache!.Id
                 GameReports.PrintIsDead();
                 throw;
             }
+
+            return;
+
+            async UniTask LoadStartingRealmAsync(CancellationToken ct)
+            {
+                try { await bootstrap.LoadStartingRealmAsync(dynamicWorldContainer!, ct); }
+                catch (RealmChangeException)
+                {
+                    if (await ShowLoadErrorPopupAsync(ct) == ErrorPopupWithRetryController.Result.RESTART)
+                        await LoadStartingRealmAsync(ct);
+                    else
+                        throw;
```

Comment:
Considering that you are aiming at making the flow "proper", throwing an exception is not an exit. Consider making the flow graceful - if the result is exit, then you should "exit" the app, not making it "dead"

Reply:
Done here: https://github.com/decentraland/unity-explorer/pull/5683/commits/311eda86f84149ea3fcd7d6a40632ae336e94961